src/ResourceManager/Profile/Commands.Profile/Context/SetAzureRMContext.cs

```diff
@@ -58,14 +56,40 @@ protected override void ProcessRecord()
                 AzureRmProfileProvider.Instance.Profile.SetContextWithCache(new AzureContext(Context.Subscription, Context.Account,
                     Context.Environment, Context.Tenant));
             }
-            else
+            else if (ParameterSetName == SubscriptionNameParameterSet)
             {
                 var profileClient = new RMProfileClient(AzureRmProfileProvider.Instance.Profile);
-                profileClient.SetCurrentContext(SubscriptionId, TenantId);
-                if (!string.IsNullOrWhiteSpace(SubscriptionName))
+                AzureSubscription subscription = null;
+                string tenantId = AzureRmProfileProvider.Instance.Profile.Context.Tenant.Id.ToString();
+
+                if (string.IsNullOrWhiteSpace(TenantId))
+                {
+                    WriteWarning(
```

Comment:
I think this should be WriteVerbose - the warning will cause some scripts to fail, depending on WarningPreference variable